node/src/reactor/main_reactor/tests.rs

```diff
@@ -1869,3 +1811,38 @@ async fn run_redelegate_bid_network() {
     // redelegated bid exists
     check_bid_existence_at_tip(&net, &charlie.1, Some(&alice.1), true);
 }
+
+// 1 -> 90
+// 2 -> 150
+#[tokio::test]
+async fn rewards_are_calculated() {
+    testing::init_logging();
+
+    let rng = &mut crate::new_rng();
+
+    // Instantiate a new chain with a fixed size.
+    const NETWORK_SIZE: usize = 5;
+    let mut chain = TestChainBuilder::new_with_size(NETWORK_SIZE)
+        .chainspec(|chainspec| {
+            chainspec.core_config.minimum_era_height = 3;
+        })
+        .build(rng);
+
+    let mut network = chain
+        .create_initialized_network(rng)
+        .await
+        .expect("network initialization failed");
+
+    let era = EraId::new(2);
+
+    network
+        .settle_on(rng, has_completed_era(era), Duration::from_secs(150))
+        .await;
+
+    let switch_block =
+        switch_block_for_era(network.first_node(), era).expect("a switch block for era 2");
+
+    for (_validator, reward) in switch_block.era_end().unwrap().rewards() {
+        assert_ne!(reward, &U512::zero());
```

Comment:
Instead of checking for a non-zero variable, would it be possible to actually check expected values for the rewards?

Reply:
It's random, so I don't really know how to check that.